packages/block-library/src/navigation-menu/menu-item-inserter.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/compose';
+import { withDispatch } from '@wordpress/data';
+import {
+	Dropdown,
+	IconButton,
+	MenuItem,
+	NavigableMenu,
+	TextControl,
+} from '@wordpress/components';
+import { useState, useMemo, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { createBlock } from '@wordpress/blocks';
+import { isURL } from '@wordpress/url';
+
+function MenuItemInserter( { insertMenuItem } ) {
+	const [ searchInput, setSearchInput ] = useState( '' );
+	const isUrlInput = useMemo( () => isURL( searchInput ), [ searchInput ] );
+	const onMenuItemClick = useCallback( () => {
+		insertMenuItem( {
+			destination: searchInput,
+		} );
+	}, [ insertMenuItem, searchInput ] );
```

Comment:
Will ever `insertMenuItem` change? Well, it's a prop passed down so React team probably recommends to include it. No action required, it's just me trying to better understand how those hooks should work :)